### PR TITLE
args: Add Arguments::from_iter

### PR DESCRIPTION
**Pull request by cgwalters**
I'm working on a test suite for https://github.com/ostreedev/ostree/
and I want to have both destructive tests and non-destructive
tests in the same binary.  I'm doing my own argument parsing for
the destructive tests and then want to use this crate for the non-destructive
ones.

### CODE REVIEW

File: src/args.rs [hunk 1]
```diff
@@ -157,6 +157,16 @@ impl Arguments {
     pub fn from_args() -> Self {
         structopt::StructOpt::from_args()
     }
+
+    /// Like from_args(), but operates on an explicit iterator and not the global arguments.
```
Comment (LukasKalbertodt) [inline src/args.rs]: Please use monospace here:

```suggestion
    /// Like `from_args()`, but operates on an explicit iterator and not the global arguments.
```

**Review by LukasKalbertodt — CHANGES_REQUESTED** [review]
Seems like a useful addition. Thanks for the PR. Only one tiny nit.

File: src/args.rs (hunk 1 above)
Comment (cgwalters) [reply src/args.rs]: Thanks, updated!

**Comment by LukasKalbertodt** [comment]
The CI failure is not due to your PR (and it's just a style thing). I will merge yours and fix that stuff on `master` later.

Also, "funny" thing I just noticed: the Travis cache was over 8GB which explains why the "downloading cache" step too forever. Ouch. Will move to GitHub actions soon anyway.

Thanks for the PR again!

**Comment by LukasKalbertodt** [comment]
Released as 0.3.0!

**Pull request merged**